### PR TITLE
Path cosmetic updates/paths

### PR DESCRIPTION
**Pull request by artemdinaburg**
Creating PR so I can comment.

### CODE REVIEW

File: osquery/core/init.cpp [hunk 1]
```diff
@@ -70,6 +70,8 @@ enum {
 
 #ifdef __linux__
 #define OSQUERY_HOME "/etc/osquery"
+#elif defined(WIN32)
+#define OSQUERY_HOME "\\ProgramData\\osquery"
 #else
```
Comment (artemdinaburg) [inline osquery/core/init.cpp]: This should really be:
```
elif defined(OSX)
...
#else
#error building on unknown OS
#endif
```

File: osquery/core/test_util.cpp [hunk 2]
```diff
@@ -28,23 +28,32 @@
...
 namespace osquery {
 
 std::string kFakeDirectory = "";
 
-#ifdef DARWIN
+#if defined(DARWIN)
 std::string kTestWorkingDirectory = "/private/tmp/osquery-tests";
+#elif defined(WIN32)
+std::string kTestWorkingDirectory =
+    (fs::temp_directory_path() / "osquery-tests").make_preferred().string();
```
Comment (artemdinaburg) [inline osquery/core/test_util.cpp]: It seems like this code would also work on DARWIN and Linux, assuming fs::temp_directory_path() returns the same thing.

File: osquery/core/test_util.cpp [hunk 3]
```diff
@@ -57,6 +66,63 @@ DECLARE_bool(disable_database);
 
 typedef std::chrono::high_resolution_clock chrono_clock;
 
+static std::string getUserId() {
+#ifdef WIN32
```
Comment (artemdinaburg) [inline osquery/core/test_util.cpp]: any easy way to remove the `#ifdef`s via cmake?

File: osquery/core/test_util.cpp [hunk 4]
```diff
@@ -57,6 +66,63 @@ DECLARE_bool(disable_database);
...
+}
+
+static std::unique_ptr<PlatformProcess> launchTestServer(const std::string &port) {
+  std::unique_ptr<PlatformProcess> server;
+
+#ifdef WIN32
+  STARTUPINFOA si = { 0 };
+  PROCESS_INFORMATION pi = { 0 };
+
+  auto argv = "python " + kTestDataPath + "/test_http_server.py --tls " + port;
```
Comment (artemdinaburg) [inline osquery/core/test_util.cpp]: Shouldn't this use `fs::path`?

File: osquery/core/test_util.cpp [hunk 5]
```diff
@@ -57,6 +66,63 @@ DECLARE_bool(disable_database);
...
+
+
+  auto drive = getEnvVar("SystemDrive");
+  std::string python_path("");
+  if (drive.is_initialized()) {
+    python_path = *drive;
+  }
+
+  // Python is installed here if provisioning script is used
+  python_path += "\\tools\python2\\python.exe";
```
Comment (artemdinaburg) [inline osquery/core/test_util.cpp]: Do we have any kind of abstracted launch process? I want to say we have/had one.

File: osquery/core/test_util.cpp [hunk 6]
```diff
@@ -74,37 +140,59 @@ void initTesting() {
...
+          .string();
   FLAGS_disable_logging = true;
   FLAGS_disable_database = true;
 
   // Tests need a database plugin.
   // Set up the database instance for the unittests.
   DatabasePlugin::setAllowOpen(true);
+
+  // Initializing database plugin does not work 
+#ifndef WIN32
```
Comment (artemdinaburg) [inline osquery/core/test_util.cpp]: Is this temporary?

File: osquery/core/init.cpp (hunk 1 above)
Comment (yying) [inline osquery/core/init.cpp]: How about FreeBSD? and whatever the other platforms they may support?

File: osquery/core/test_util.cpp (hunk 5 above)
Comment (yying) [inline osquery/core/test_util.cpp]: Ours is more tailored towards launching extensions/workers.

File: osquery/core/test_util.cpp (hunk 6 above)
Comment (yying) [inline osquery/core/test_util.cpp]: Yes, perhaps should be removed in the commit...

**Pull request closed without merging**